File: bot.py
import discord
import os
from discord.ext import commands

# Set up the bot and intents
my_secret = os.environ['TOKEN']
intents = discord.Intents.default()
intents.message_content = True
bot = commands.Bot(command_prefix='!', intents=intents)

# Define the announcement channel ID and the announcement role ID
ANNOUNCEMENT_CHANNEL_ID = 1208377063870824458  # Replace with the actual channel ID
ANNOUNCEM_CHANNEL_ID = 1208377024981237801  # Replace with the actual channel ID
ANNOUNCEMENT_ROLE_ID = 1208379129523605584  # Replace with the actual role ID you want to mention

@bot.command()
async def feedback(ctx, *, user_feedback: str):
    # Replace 'your_feedback_channel_id' with the actual channel ID where you want to receive feedback
    feedback_channel = bot.get_channel(1262021731292418119)

    if feedback_channel:
        await feedback_channel.send(f"New feedback from {ctx.author.mention}:\n{user_feedback}")
        await ctx.send("Thank you for your feedback! It has been submitted.")
    else:
        await ctx.send("Feedback channel not found. Please set up a valid feedback channel.")

@bot.command()
@commands.has_permissions(administrator=True)  # Check if the user has Administrator permissions
async def announce(ctx, *, message: str):
    announcement_channel = bot.get_channel(ANNOUNCEMENT_CHANNEL_ID)
    announcement_role = ctx.guild.get_role(ANNOUNCEMENT_ROLE_ID)  # Get the role object

    if announcement_channel:
        if announcement_role:
            # Mention the role and send the announcement message
            await announcement_channel.send(f"{announcement_role.mention} {message}")
            await ctx.send("Announcement has been sent.")
        else:
            await ctx.send("Announcement role not found. Please set up a valid role.")
    else:
        await ctx.send("Announcement channel not found. Please set up a valid announcement channel.")

# Handle permission error
@announce.error
async def announce_error(ctx, error):
    if isinstance(error, commands.MissingPermissions):
        await ctx.send("You don't have the required permissions to use this command.")

@bot.command()
@commands.has_permissions(administrator=True)  # Check if the user has Administrator permissions
async def announcement(ctx, *, message: str):
    announcement_channel = bot.get_channel(ANNOUNCEM_CHANNEL_ID)
    announcement_role = ctx.guild.get_role(ANNOUNCEMENT_ROLE_ID)  # Get the role object

    if announcement_channel:
        if announcement_role:
            # Mention the role and send the announcement message
            await announcement_channel.send(f"{announcement_role.mention} {message}")
            await ctx.send("Announcement has been sent.")
        else:
            await ctx.send("Announcement role not found. Please set up a valid role.")
    else:
        await ctx.send("Announcement channel not found. Please set up a valid announcement channel.")

# Handle permission error
@announcement.error
async def announce_error(ctx, error):
    if isinstance(error, commands.MissingPermissions):
        await ctx.send("You don't have the required permissions to use this command.")

# Run the bot
bot.run(my_secret)

File: bot.lua
local discordia = require('discordia')
local client = discordia.Client()

-- Define the announcement role IDs
local ANNOUNCEMENT_ROLE_ID = 1208379129523605584
local LEAKS_ROLE_ID = 1208379179134091334

client:on('messageCreate', function(message)
    if message.author == client.user then return end

    -- Command: !feedback
    if message.content:match('^!feedback ') then
        local feedback = message.content:sub(10) -- Extract feedback text
        local feedback_channel = client:getChannel(1262021731292418119) -- Replace with your feedback channel ID

        if feedback_channel then
            feedback_channel:send("New feedback from " .. message.author.mentionString .. ":\n" .. feedback)
            message.channel:send("Thank you for your feedback! It has been submitted.")
        else
            message.channel:send("Feedback channel not found. Please set up a valid feedback channel.")
        end

    -- Command: !announce
    elseif message.content:match('^!announce ') and message.member:hasRole(ANNOUNCEMENT_ROLE_ID) then
        local content = message.content:sub(10) -- Extract command content
        local args = content:match("^(.-) (.+)$") -- Split into channel and message
        if args then
            local channel_id, msg = args:match("^(%d+) (.+)$")
            local channel = client:getChannel(tonumber(channel_id))
            local announcement_role = message.guild:getRole(ANNOUNCEMENT_ROLE_ID)

            if channel then
                if announcement_role then
                    channel:send(announcement_role.mentionString .. " " .. msg)
                    message.channel:send("Announcement has been sent.")
                else
                    message.channel:send("Announcement role not found. Please set up a valid role.")
                end
            else
                message.channel:send("Announcement channel not found. Please set up a valid channel.")
            end
        else
            message.channel:send("Invalid command format. Use !announce <channel_id> <message>")
        end

    -- Command: !leaks
    elseif message.content:match('^!leaks ') and message.member:hasRole(LEAKS_ROLE_ID) then
        local content = message.content:sub(7) -- Extract command content
        local announcement_channel = client:getChannel(1208377063870824458) -- Replace with your leaks channel ID
        local announcement_role = message.guild:getRole(LEAKS_ROLE_ID)

        if announcement_channel then
            if announcement_role then
                announcement_channel:send(announcement_role.mentionString .. " " .. content)
                message.channel:send("Announcement has been sent.")
            else
                message.channel:send("Announcement role not found. Please set up a valid role.")
            end
        else
            message.channel:send("Announcement channel not found. Please set up a valid announcement channel.")
        end
    end
end)

client:run('Bot [TOKEN]')
